rs/sns/governance/src/neuron.rs

```diff
@@ -24,6 +24,9 @@ pub const MAX_LIST_NEURONS_RESULTS: u32 = 100;
 /// The default voting_power_percentage_multiplier applied to a neuron.
 pub const DEFAULT_VOTING_POWER_PERCENTAGE_MULTIPLIER: u64 = 100;
 
+// The max number of reward_events_to_neuron_reward_e8s per neuron
+pub const MAX_KEEP_REWARD_EVENTS_TO_NEURON_REWARD_E8S_PER_NEURON: u32 = 5;
```

Comment:
If it's set to 1 the caller would need to sync at least 2/3/4 times per day to make sure to get the data. If a caller only syncs once per day then it might miss the reward-event because a sns-reward-event can happen in between the different list_neurons calls and the caller will not know if the neurons (in the list_neurons chunks after the sns-reward-event) got their reward for the previous reward-event or if they were not eligible for rewards since the sns-reward-event just pruned them. So since here it is > 1, the caller canister can call once a day, and start from the second-to-earliest reward-event in the map and that way the reward-event will for sure be there on all neurons that voted for that second-to-earliest reward-event even if a sns-reward-event happens in between the list_neurons calls as long as the list_neurons calls take less than a day to go through all the neurons. 

Another reason is if a caller canister needs maintenance for a day its better if it can catch up. I think one day window to sync the data is too cramped.

Combining these reasons gives a value of 3 for the constant, and then add 2 for good measure to make 5. That's how I think about it. I am cool with setting the value of the constant to 3, if you think that is better.